Sources/Testing/Parameterization/CustomTestArgument.swift

```diff
@@ -0,0 +1,154 @@
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+#if canImport(Foundation)
+private import Foundation
+#endif
+
+/// A protocol for customizing how arguments passed to parameterized tests are
+/// encoded, which is used to match against when running specific arguments.
+@_spi(ExperimentalParameterizedTesting)
+public protocol CustomTestArgumentEncodable: Sendable {
+  /// Encode this test argument.
+  ///
+  /// - Parameters:
+  ///   - encoder: The encoder to write data to.
+  ///
+  /// - Throws: Any error encountered during encoding.
+  ///
+  /// The encoded form of a test argument should be stable and unique to allow
+  /// re-running specific test cases of a parameterized test function. For
+  /// optimal performance, large values which are not necessary to uniquely
+  /// identify the test argument later should be omitted. Values encoded do not
+  /// need to be human-readable.
+  ///
+  /// By default, the testing library checks whether a test argument conforms to
+  /// `Encodable` and encodes it using `encodeTestArgument(to:)` if it does.
+  /// This is sufficient for many types, but for some types the `Encodable`-
+  /// provided representation may contain large data payloads that cause poor
+  /// performance, are not stable and unique, or are otherwise deemed unsuitable
+  /// for testing. If the type of the argument does not conform to `Encodable`
+  /// but it does conform to `Identifiable` and its associated `ID` type
+  /// conforms to `Encodable`, the value of calling its `id` property is used as
+  /// the encoded representation.
+  ///
+  /// It is possible that neither of the approaches for encoded representation
+  /// described above are sufficient. If the type of the argument is made to
+  /// conform to ``CustomTestArgumentEncodable``, then the encoded
+  /// representation formed by calling this method is used.
+  func encodeTestArgument(to encoder: some Encoder) throws
+}
+
+extension Test.Case.Argument.ID {
+  /// Initialize this instance with an ID for the specified test argument.
+  ///
+  /// - Parameters:
+  ///   - value: The value of a test argument for which to get an ID.
+  ///   - parameter: The parameter of the test function to which this argument
+  ///     value was passed.
+  ///
+  /// - Returns: `nil` if an ID cannot be formed from the specified test
+  ///   argument value.
+  ///
+  /// - Throws: Any error encountered while attempting to encode `value`.
+  ///
+  /// This function is not part of the public interface of the testing library.
+  ///
+  /// ## See Also
+  ///
+  /// - ``CustomTestArgumentEncodable``
+  init?(identifying value: some Sendable, parameter: Test.ParameterInfo) throws {
+#if canImport(Foundation)
+    guard Configuration.current?.isTestArgumentEncodingEnabled ?? false else {
+      return nil
+    }
+
+    func _customArgumentWrapper(for value: some CustomTestArgumentEncodable) -> some Encodable {
```

Comment:
You can shuffle this down right into the `if` branch that uses it. And drop the leading underscore.

Reply:
Can't because I am using [SE-0380 (If and switch expressions)](https://github.com/apple/swift-evolution/blob/main/proposals/0380-if-switch-expressions.md). Dropped the leading underscore